### PR TITLE
Dropping QT4 and Eigen is now required (and included using FetchContent)

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

Including eigen in DGtal (no more optional):
* we use FetchContent to download the proper version at cmake time
*  `EIGEN_ROOT` can be used to specify alternative eigen install 

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [ ] No warning raised in Debug mode.
- [ ] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

**Comment by dcoeurjo** [comment]
@JacquesOlivierLachaud @kerautret @troussil could you please test this PR on your system ?

**Comment by kerautret** [comment]
Yes I look it

**Comment by kerautret** [comment]
All looks fine both from my computer on MacOS and also tested on linux servers with QT options. 👌🏻

**Comment by troussil** [comment]
it seems fine (last LTS Ubuntu, even when QGLViewer and examples are enable)

File: .github/workflows/buildAndDocumentation-PR.yml [hunk 1]
```diff
@@ -32,11 +32,12 @@ jobs:
        if: matrix.os == 'ubuntu-latest'
        run: |
            sudo apt-get update
-           sudo apt-get install zsh libqglviewer-dev-qt5 libboost-dev libeigen3-dev ninja-build libhdf5-serial-dev libboost-dev libcairo2-dev libgmp-dev libfftw3-dev libinsighttoolkit4-dev
+           sudo apt-get install zsh libqglviewer-dev-qt5 libboost-dev ninja-build libhdf5-serial-dev libboost-dev libcairo2-dev libgmp-dev libfftw3-dev libinsighttoolkit4-dev
 
      - name: Install macOS deps
        if: matrix.os == 'macOS-latest'
-       run: brew install boost ninja eigen gmp libomp
+       run: brew install boost ninja gmp
```
Comment (dcoeurjo) [inline .github/workflows/buildAndDocumentation-PR.yml:39]:
```suggestion
       run: brew install boost ninja gmp libomp 
```

**Review by dcoeurjo — COMMENTED** [review]
Ok

**Comment by dcoeurjo** [comment]
Thx

File: .github/workflows/buildAndDocumentation.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  # Documentatin (build, check and deploy)
+  Documentation:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up
+      run:  sudo apt-get install libboost-dev graphviz texlive doxygen
```
Comment (dcoeurjo) [inline .github/workflows/buildAndDocumentation.yml:95]:
```suggestion
      run:  sudo apt-get update && sudo apt-get install libboost-dev graphviz texlive doxygen
```

**Review by dcoeurjo — COMMENTED** [review]
Editing

**Pull request merged**